website/docs/r/host_virtual_switch.html.markdown

```diff
@@ -0,0 +1,173 @@
+For an overview on vSphere networking concepts, see [this
+page][ref-vsphere-net-concepts].
+
+[host-port-group]: /docs/providers/vsphere/r/host_port_group.html
+[ref-vsphere-net-concepts]: https://docs.vmware.com/en/VMware-vSphere/6.5/com.vmware.vsphere.networking.doc/GUID-2B11DBB8-CB3C-4AFF-8885-EFEA0FC562F4.html
+
+## Example Usages
+
+**Create a virtual switch with one active and one standby NIC:**
+
+```hcl
+data "vsphere_datacenter" "datacenter" {
+  name = "dc1"
+}
+
+data "vsphere_host" "host" {
+  name          = "esxi1"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.id}"
+
+  network_adapters = ["vmnic0", "vmnic1"]
+
+  active_nics  = ["vmnic0"]
+  standby_nics = ["vmnic1"]
+}
+```
+
+**Create a virtual switch with extra networking policy options:**
+
+```hcl
+data "vsphere_datacenter" "datacenter" {
+  name = "dc1"
+}
+
+data "vsphere_host" "host" {
+  name          = "esxi1"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.id}"
+
+  network_adapters = ["vmnic0", "vmnic1"]
+
+  active_nics    = ["vmnic0"]
+  standby_nics   = ["vmnic1"]
+  teaming_policy = "failover_explicit"
+
+  allow_promiscuous      = false
+  allow_forged_transmits = false
+  allow_mac_changes      = false
+
+  shaping_enabled           = true
+  shaping_average_bandwidth = 50000000
+  shaping_peak_bandwidth    = 100000000
+  shaping_burst_size        = 1000000000
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (String, required, forces new resource) The name of the virtual switch.
+* `host_system_id` - (String, required, forces new resource) The managed object
+  ID of the host to set the virtual switch up on. 
+* `mtu` - (Integer, optional) The maximum transmission unit (MTU) for the virtual
+  switch. Default: `1500`.
+* `number_of_ports` - (Integer, optional) The number of ports to create with
+  this virtual switch. Default: `128`.
+
+~> **NOTE:** Changing the port count requires a reboot of the host. Terraform
+will not restart the host for you.
+
+### Bridge Options
+
+The following arguments are related to how the virtual switch binds to physical
+NICs:
+
+* `network_adapters` - (Array of strings, required) The network interfaces to
+  bind to the bridge.
+* `beacon_interval` - (Integer, optional) The interval, in seconds, that a NIC
+  beacon packet is sent out. This can be used with
+  [`check_beacon`](#check_beacon) to offer link failure capability beyond link
+  status only. Default: `1`.
+* `link_discovery_operation` - (String, optional) Whether to `advertise` or
+  `listen` for link discovery traffic. Default: `listen`.
+* `link_discovery_protocol` - (String, optional) The discovery protocol type.
```

Comment:
Maybe this is consistent with the other docs in vmware land - but in other providers I've seen:

```
* `foo` (Optional) Setting this to `blah` does something awesome. Changing this forces a new resource to be created
```

again, not a blocker but should we be consolidating on this?

Reply:
I think I've seen a number of styles, not just in this provider but in other providers as well.

At this point in time, I think there'd need to be a bigger push towards consistency and one singular convention in the provider docs here in general. For now I think I will leave it and then add a doc review as a longer term TODO to make things consistent.